mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -388,19 +388,21 @@ class Word2Vec extends Serializable with Logging {
         val syn0Local = model._1
         val syn1Local = model._2
         // Only output modified vectors.
-        Iterator.tabulate(vocabSize) { index =>
+        val modifiedSize = syn0Modify.count(_ > 0) + syn1Modify.count(_ > 0)
+        val modified = new Array[(Int, Array[Float])](modifiedSize)
+        var mi = 0
+        (0 until vocabSize).foreach { index =>
```

Comment:
I think this makes sense if you've identified it as a bottleneck (I feel like another PR tried to do this ...) but doesn't this change the order of the output? maybe it doesn't matter. But it's interleaved now.